### PR TITLE
EOS-14340: cortx-py-utils rpm installation does not install dependencies

### PR DESCRIPTION
**Pull request by mssawant**
Following packages are required by cortx-py-utils but are not installed as part of
the rpm installation. Components using cortx-py-utils fail due to these
dependencies and require to handle them separately.
```
aiohttp==3.6.1
elasticsearch-dsl==6.4.0
python-consul==1.1.0
schematics==2.1.0
toml==0.10.0
```
Solutions:
- Add  a post install script to install the dependencies.
  This script can be specified to setup.py bdist_rpm command
  as follows,
```
   python3.6 setup.py bdist_rpm --post-install utils-post-install
```
  Built rpm can then be installed along with dependencies,

```
[root@ssc-vm-c-0552 py-utils]# rpm -ivh dist/cortx-py-utils-1.0.0-1.noarch.rpm
Preparing...                          ################################# [100%]
Updating / installing...
   1:cortx-py-utils-1.0.0-1           ################################# [100%]
WARNING: Running pip install with root privileges is generally not a good idea. Try `__main__.py install --user` instead.
Collecting aiohttp==3.6.1
  Downloading https://files.pythonhosted.org/packages/e0/cb/5fa1d7722488995358d6092a7d2ccceb9011ab78842ccd336fbb5ec0d878/aiohttp-3.6.1-cp36-cp36m-manylinux1_x86_64.whl (1.2MB)
    100% |████████████████████████████████| 1.2MB 711kB/s
Requirement already satisfied: async-timeout<4.0,>=3.0 in /usr/local/lib/python3.6/site-packages (from aiohttp==3.6.1)
Requirement already satisfied: idna-ssl>=1.0; python_version < "3.7" in /usr/local/lib/python3.6/site-packages (from aiohttp==3.6.1)
Requirement already satisfied: typing-extensions>=3.6.5; python_version < "3.7" in /usr/local/lib/python3.6/site-packages (from aiohttp==3.6.1)
Requirement already satisfied: chardet<4.0,>=2.0 in /usr/local/lib/python3.6/site-packages (from aiohttp==3.6.1)
Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib64/python3.6/site-packages (from aiohttp==3.6.1)
Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.6/site-packages (from aiohttp==3.6.1)
Requirement already satisfied: multidict<5.0,>=4.5 in /usr/local/lib64/python3.6/site-packages (from aiohttp==3.6.1)
Requirement already satisfied: idna>=2.0 in /usr/local/lib/python3.6/site-packages (from idna-ssl>=1.0; python_version < "3.7"->aiohttp==3.6.1)
Installing collected packages: aiohttp
  Found existing installation: aiohttp 3.6.2
    Uninstalling aiohttp-3.6.2:
      Successfully uninstalled aiohttp-3.6.2
Successfully installed aiohttp-3.6.1
WARNING: Running pip install with root privileges is generally not a good idea. Try `__main__.py install --user` instead.
Collecting elasticsearch-dsl==6.4.0
  Downloading https://files.pythonhosted.org/packages/4f/73/580e648ab72a19c26c815931ce2b2dda67454f557e72ce7ae90fd3b7887d/elasticsearch_dsl-6.4.0-py2.py3-none-any.whl (49kB)
    100% |████████████████████████████████| 51kB 1.0MB/s
Collecting elasticsearch<7.0.0,>=6.0.0 (from elasticsearch-dsl==6.4.0)
  Downloading https://files.pythonhosted.org/packages/78/1b/498aa74c244c5c9d578c1a1a2f16b0db61ac178ec4f1a33d3785f86a052d/elasticsearch-6.8.1-py2.py3-none-any.whl (75kB)
    100% |████████████████████████████████| 81kB 2.6MB/s
Requirement already satisfied: python-dateutil in /usr/local/lib/python3.6/site-packages (from elasticsearch-dsl==6.4.0)
Requirement already satisfied: six in /usr/lib64/python3.6/site-packages (from elasticsearch-dsl==6.4.0)
Requirement already satisfied: urllib3>=1.21.1 in /usr/local/lib/python3.6/site-packages (from elasticsearch<7.0.0,>=6.0.0->elasticsearch-dsl==6.4.0)
Installing collected packages: elasticsearch, elasticsearch-dsl
  Found existing installation: elasticsearch 7.9.1
    Uninstalling elasticsearch-7.9.1:
      Successfully uninstalled elasticsearch-7.9.1
  Found existing installation: elasticsearch-dsl 7.3.0
    Uninstalling elasticsearch-dsl-7.3.0:
      Successfully uninstalled elasticsearch-dsl-7.3.0
Successfully installed elasticsearch-6.8.1 elasticsearch-dsl-6.4.0
WARNING: Running pip install with root privileges is generally not a good idea. Try `__main__.py install --user` instead.
Requirement already satisfied: python-consul==1.1.0 in /usr/local/lib/python3.6/site-packages
Requirement already satisfied: six>=1.4 in /usr/lib64/python3.6/site-packages (from python-consul==1.1.0)
Requirement already satisfied: requests>=2.0 in /usr/local/lib/python3.6/site-packages (from python-consul==1.1.0)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.6/site-packages (from requests>=2.0->python-consul==1.1.0)
Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.6/site-packages (from requests>=2.0->python-consul==1.1.0)
Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/site-packages (from requests>=2.0->python-consul==1.1.0)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.6/site-packages (from requests>=2.0->python-consul==1.1.0)
WARNING: Running pip install with root privileges is generally not a good idea. Try `__main__.py install --user` instead.
Requirement already satisfied: schematics==2.1.0 in /usr/local/lib/python3.6/site-packages
WARNING: Running pip install with root privileges is generally not a good idea. Try `__main__.py install --user` instead.
Collecting toml==0.10.0
  Downloading https://files.pythonhosted.org/packages/a2/12/ced7105d2de62fa7c8fb5fce92cc4ce66b57c95fb875e9318dba7f8c5db0/toml-0.10.0-py2.py3-none-any.whl
Installing collected packages: toml
  Found existing installation: toml 0.10.1
    Uninstalling toml-0.10.1:
      Successfully uninstalled toml-0.10.1
Successfully installed toml-0.10.0
```

### CODE REVIEW

File: py-utils/utils-post-uninstall [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+PYTHON=python3.6 
+
+sudo $PYTHON -m pip uninstall aiohttp <<< $(echo 'y')
+sudo $PYTHON -m pip uninstall elasticsearch-dsl <<< $(echo 'y')
+sudo $PYTHON -m pip uninstall python-consul <<< $(echo 'y')
+sudo $PYTHON -m pip uninstall schematics <<< $(echo 'y')
+sudo $PYTHON -m pip uninstall toml <<< $(echo 'y')
```
Comment (navalkp) [inline py-utils/utils-post-uninstall:9]: May be in another patch having a requirement.txt and setup.py , utils-post-install and utils-post-uninstall referring to it will have only one place to add a new package.

File: py-utils/utils-post-install [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+PYTHON=python3.6 
+
+sudo $PYTHON -m pip install 'aiohttp==3.6.1'
+sudo $PYTHON -m pip install 'elasticsearch-dsl==6.4.0'
+sudo $PYTHON -m pip install 'python-consul==1.1.0'
+sudo $PYTHON -m pip install 'schematics==2.1.0'
+sudo $PYTHON -m pip install 'toml==0.10.0'
```
[the remark this answers is not in the document]
Comment (mssawant) [reply py-utils/utils-post-install:9]: Yeah, I have a requirement.txt from another patch that I used for Makefile and specs file. Let's add it and use it here.

File: py-utils/utils-post-uninstall (hunk 1 above)
Comment (mssawant) [reply py-utils/utils-post-uninstall:9]: Yeah let's do this now.

**Review by shrimetta — APPROVED** [review]

File: py-utils/utils-post-uninstall (hunk 1 above)
Comment (kupranay) [reply py-utils/utils-post-uninstall:9]: I would suggest lets do it correctly. "Tomorrow never comes :)"

**Review by kupranay — APPROVED** [review]

File: py-utils/utils-post-uninstall (hunk 1 above)
Comment (mssawant) [reply py-utils/utils-post-uninstall]: Done.

**Comment by mssawant** [comment]
@navalkp , the command now changes to `python3.6 setup.py bdist_rpm --post-install utils-post-install --pre-uninstall utils-pre-uninstall`

File: py-utils/setup.py [hunk 3]
```diff
@@ -60,7 +60,8 @@
         ]
       },
       data_files = [ ('/var/lib/cortx/ha/specs', specs),
-                     ('/var/lib/cortx/ha', ['src/utils/ha/hac/args.yaml', 'src/utils/ha/hac/re_build.sh'])],
+                     ('/var/lib/cortx/ha', ['src/utils/ha/hac/args.yaml', 'src/utils/ha/hac/re_build.sh']),
+                     ('/opt/seagate/cortx/utils/conf', ['requirement.txt'])],
```
Comment (navalkp) [inline py-utils/setup.py:64]: Can we populate install_require from requirement.txt

Comment (navalkp) [reply py-utils/setup.py:64]: Please update py-utils/README.md also with commandline to build rpm

Comment (mssawant) [reply py-utils/setup.py:64]: >Can we populate install_require from requirement.txt

yes why not, good idea! Let me update the patch.

Comment (mssawant) [reply py-utils/setup.py]: Done.

**Review by navalkp — APPROVED** [review]

**Review by ajaysrivas — APPROVED** [review]

**Review by ajay-paratmandali — APPROVED** [review]

**Review by sachinpunadikar — APPROVED** [review]
+1
